src/native/minipal/cpufeatures.c

```diff
@@ -220,54 +220,35 @@ int minipal_getcpufeatures(void)
-                                                result |= XArchIntrinsicConstants_Avx512f_vl;
-                                                isAVX512_VLSupported = true;
-                                            }
-
-                                            if ((cpuidInfo[CPUID_EBX] & (1 << 30)) != 0)                        // AVX512BW
-                                            {
-                                                result |= XArchIntrinsicConstants_Avx512bw;
-                                                if (isAVX512_VLSupported)                                       // AVX512BW_VL
-                                                {
-                                                    result |= XArchIntrinsicConstants_Avx512bw_vl;
-                                                }
-                                            }
-
-                                            if ((cpuidInfo[CPUID_EBX] & (1 << 28)) != 0)                        // AVX512CD
-                                            {
-                                                result |= XArchIntrinsicConstants_Avx512cd;
-                                                if (isAVX512_VLSupported)                                       // AVX512CD_VL
-                                                {
-                                                    result |= XArchIntrinsicConstants_Avx512cd_vl;
-                                                }
-                                            }
+                                            // While the AVX-512 ISAs can be individually lit-up, they really
+                                            // need F, BW, CD, DQ, and VL to be fully functional without adding
+                                            // significant complexity into the JIT. Additionally, unlike AVX/AVX2
+                                            // there was never really any hardware that didn't provide all 5 at
+                                            // once, with the notable exception being Knight's Landing which
+                                            // provided a similar but not quite the same feature.
 
-                                            if ((cpuidInfo[CPUID_EBX] & (1 << 17)) != 0)                        // AVX512DQ
-                                            {
-                                                result |= XArchIntrinsicConstants_Avx512dq;
-                                                if (isAVX512_VLSupported)                                       // AVX512DQ_VL
-                                                {
-                                                    result |= XArchIntrinsicConstants_Avx512dq_vl;
-                                                }
-                                            }
+                                            result |= XArchIntrinsicConstants_Evex;
+                                            result |= XArchIntrinsicConstants_Avx512f;
```

Comment:
If we bake the JIT policy of requiring all these features together at the same time into the minipal, we can fold all these bits into just a single bit `XArchIntrinsicConstants_Avx512f`.

Reply:
The code in JIT/EE interface https://github.com/dotnet/runtime/pull/103241/files#diff-eba17a72b6caa8b52b196f3b84bfd20dfeaf39a89f410943a692d1f33f569f3bR1366 does the same thing. What's the benefit of doing this in the minipal too?

Reply:
> If we bake the JIT policy of requiring all these features together at the same time into the minipal, we can fold all these bits into just a single bit XArchIntrinsicConstants_Avx512f.

Right, and I think that's desirable long term. I've updated the PR to fold it down to `Avx512` and `Avx512Vbmi`

> What's the benefit of doing this in the minipal too?

It's helping enforce the consistency everywhere that consumes the CPUID bits.

What I ended up finding is that the `implication` defined in `InstructionSetDesc` don't work well with recursive implication sets. This is because it basically does a naive ordering and so what happened is that `Avx512F` would be supported, which meant that `Avx512BW/CD/DQ` had their requirements met. Then at end when processing the recursive implications, if one of `BW/CD/DQ/VL` wasn't supported, `F` would be unset but the others wouldn't leading to a torn view.

While handling this in codeman is sufficient for the JIT, it doesn't necessarily mean we followed suite in NAOT or other scenarios as they could get disconnected. So simply enforcing this at the VM level makes it a bit easier and ensures everywhere has a consistent view of what we require to be supported together.

More ideally we'd also clean this up at the `InstructionSet_*` level too as we don't need `instructionset64bit` (this is implied by the target architecture bitness), we don't need `instructionset Avx512F_VL` (this could be compressed down to just `Avx512F`), etc. I think doing this when we're ready for the next R2R breaking change would be good, as it should free up bits and reduce the overall complexity of the JIT by just requiring a little bit of handling in the `lookupIntrinsicId` function instead.

Reply:
> I think doing this when we're ready for the next R2R breaking change would be good

We just took a R2R breaking change a few days ago, so if you would like to cleanup the `CorInfoInstructionSet` bits, it is a good time now.